src/spec-node/imageMetadata.ts

```diff
@@ -102,14 +113,37 @@ const replaceProperties = [
 interface UpdatedConfigProperties {
 	customizations?: Record<string, any[]>;
 	entrypoints?: string[];
-	onCreateCommands?: (string | string[])[];
-	updateContentCommands?: (string | string[])[];
-	postCreateCommands?: (string | string[])[];
-	postStartCommands?: (string | string[])[];
-	postAttachCommands?: (string | string[])[];
+	onCreateCommands?: LifecycleCommand[];
+	updateContentCommands?: LifecycleCommand[];
+	postCreateCommands?: LifecycleCommand[];
+	postStartCommands?: LifecycleCommand[];
+	postAttachCommands?: LifecycleCommand[];
 	shutdownAction?: 'none' | 'stopContainer' | 'stopCompose';
 }
 
+export function lifecycleCommandOriginMapFromMetadata(metadata: ImageMetadataEntry[]): LifecycleHooksInstallMap {
```

Comment:
One alternative would be to change MergedDevContainerConfig in `imageMetadata.ts` to remember the origin (the merged config is already passed around).